### ISSUE TITLE
Integrity option in importScripts

### ISSUE THREAD

**Issue by matthewbauer** [w3c/webappsec#454]
Is there a way to get integrity checking within web workers? It would be nice to have an option analogous to the "integrity" attribute for "[importScripts](https://html.spec.whatwg.org/multipage/workers.html#dom-workerglobalscope-importscripts)" used within web workers.

I would propose extending the current WorkerGlobalScope like:

```
[Exposed=Worker]
partial interface WorkerGlobalScope {
  void importScripts(DOMString... urls, DOMString... integrity);
};
```

**Comment by annevk** [comment]
I don't see how that overload could work. We'll need a new method that takes options. Or overload `DOMString` with `Request` objects maybe.

**Comment by matthewbauer** [comment]
Right that overload doesn't make sense. But, is some specification of this worth the effort? I believe that "importScripts" was originally just added as a convenient way to import a script without DOM access until ES6 module loading is worked out. It definitely is a smaller use case than with script and link.

**Comment by annevk** [comment]
Yeah, modules might be the answer. Though modules and workers require some work. Since if you want to setup a loader we'll need a way for a worker to execute multiple scripts in order, similar to what HTML can do.

**Comment by matthewbauer** [comment]
Okay, so something like this, changing DOMString to RequestInfo:

```
partial interface WorkerGlobalScope {
  void importScripts(RequestInfo... urls);
}
```

Allowing for compatibility with the old ```importScripts("abc.js");``` while enabling something like:

```js
importScripts(new Request("abc.js", {integrity: "sha256-"}));
```

We'd want to make sure that the [RequestInit](https://fetch.spec.whatwg.org/#requestinit) object couldn't be used for any malicious behavior (especially with ```credentials: "include"```).

**Comment by annevk** [comment]
A modification how? `importScripts()` already includes credentials.

**Comment by matthewbauer** [comment]
Sorry that was confusing. I just meant that this would change how "importScripts" is implemented and add more flexibility that might not be wanted. That is there are some things that make sense when you are "fetch"ing a resource but less sense when you are importing a script like headers, body, etc.

**Comment by indolering** [comment]
I'm guessing an object overload won't work due to backwards compatibility issues?

We could append the SRI after a hash:

    importScripts('https://cdnjs.cloudflare.com/ajax/libs/jquery/3.1.1/jquery.js#sha256-16cdPddA6VdVInumRGo6IbivbERE8p7CQR3HzTBuELA=');

Firefox and Chrome load it just fine.  It would be better if we could parameterize the request but this will work....

**Comment by annevk** [comment]
Also, should we not first add integrity to workers?

**Comment by annevk** [comment]
An overload would work btw, but I'm not sure it's worth doing given JS modules.

**Comment by indolering** [comment]
> I'm not sure it's worth doing given JS modules.

I've been having a hard time getting CDN's to display the SRI hash by default, perhaps this would be a worthwhile enough hack that it could be supported higher up?  It would allow transparent SRI support in a lot of interfaces that might not otherwise get it.
